heartbeat/podman-etcd

```diff
@@ -1029,6 +1029,48 @@ get_peer_node_name() {
 	crm_node -l | awk '{print $2}' | grep -v "$NODENAME"
 }
 
+# Calculate the count of truly active resources by excluding those being stopped.
+# According to Pacemaker documentation, during "Post-notification (stop) /
+# Pre-notification (start)" transitions, the true active resource count should be:
+#   Active resources = $OCF_RESKEY_CRM_meta_notify_active_resource
+#                      minus $OCF_RESKEY_CRM_meta_notify_stop_resource
+# This handles the case where a resource appears in both the active and stop lists
+# during rapid restart scenarios (e.g., process crash recovery).
+get_truly_active_resources_count() {
+	local active_list="$OCF_RESKEY_CRM_meta_notify_active_resource"
+	local stop_list="$OCF_RESKEY_CRM_meta_notify_stop_resource"
+	local truly_active=""
+
+	# If no active resources, return 0
+	if [ -z "$active_list" ]; then
+		echo "0"
+		return
+	fi
+
+	# If no resources being stopped, return count of active resources
+	if [ -z "$stop_list" ]; then
+		echo "$active_list" | wc -w
+		return
+	fi
+
+	# Filter out resources that are being stopped from the active list
+	for resource in $active_list; do
+		local is_stopping=0
+		for stop_resource in $stop_list; do
+			if [ "$resource" = "$stop_resource" ]; then
+				is_stopping=1
+				break
+			fi
+		done
+		if [ $is_stopping -eq 0 ]; then
+			truly_active="$truly_active $resource"
+		fi
+	done
```

Comment:
Considering that:
* we return the count of words in `$truly_active`
* we know for a fact that our pool will be made of only 2 resources (i.e. 1 active, 1 stop, or 2 actives, 0 stop, etc.)
* the problems with `-z` test in the previous comment

It seems to me that we could simplify the function with some math, something like

```
get_truly_active_resources_count() {
    local active_count stop_count

    active_count=$(echo "$OCF_RESKEY_CRM_meta_notify_active_resource" | wc -w)
    stop_count=$(echo "$OCF_RESKEY_CRM_meta_notify_stop_resource" | wc -w)

    echo $((active_count - stop_count))
}


```